src/session/session-utils.c

```diff
@@ -70,23 +63,74 @@ read_authorize_response (const char *what)
        exit (0);
     }
 
+  /* make room for and add null terminator */
+  message = realloc (message, len + 1);
```

Comment:
It might make sense to just allocate (and `nul`-fill) that extra byte in `cockpit_frame_read()`.  You are its only remaining user (outside of tests).

Reply:
Good idea, thanks! Done.